src/lsstseries/analysis/structure_function/bauer_2009b/calculator.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+
+from lsstseries.analysis.structure_function.base_argument_container import StructureFunctionArgumentContainer
+from lsstseries.analysis.structure_function.base_calculator import StructureFunctionCalculator
+
+PI_OVER_2 = np.pi / 2.0
+
+
+class Bauer2009BStructureFunctionCalculator(StructureFunctionCalculator):
+    """This class implements the structure function calculation described in
+    Eqn. 5 of Bauer et al. 2009, 2009ApJ...696.1241B [https://arxiv.org/abs/0902.4103]
+
+    `SF(tau) = sqrt( (pi/2) * mean(abs(delta_flux))^2 - mean(err^2) )`
+
+    Additional references:
+    Graham et al. 2014MNRAS.439..703G [https://arxiv.org/abs/1401.1785]
+    """
+
+    def calculate(self):
+        err2_values = []
+        for lc_idx in range(len(self._time)):
+            lc_times = self._time[lc_idx]
+            lc_fluxes = self._flux[lc_idx]
+            lc_errors = self._err[lc_idx]
+
+            # mask out any nan values
+            t_mask = np.isnan(lc_times)
+            f_mask = np.isnan(lc_fluxes)
+            e_mask = np.isnan(lc_errors)  # always mask out nan errors?
+            lc_mask = np.logical_or(t_mask, f_mask, e_mask)
+
+            lc_times = lc_times[~lc_mask]
+            lc_fluxes = lc_fluxes[~lc_mask]
+            lc_errors = lc_errors[~lc_mask]
+
+            # compute d_times (difference of times) and
+            # d_fluxes (difference of magnitudes, i.e., fluxes) for all gaps
+            # d_times - difference of times
```

Comment:
This comment is a repeat of the information on line 36.

Reply:
Fixed.

Reply:
Fixed.